gauntlet/packages/gauntlet-solana-contracts/src/lib/contracts.ts

```diff
@@ -36,7 +38,27 @@ const getContractSchema = (name: CONTRACT_LIST, version: string) => {
   return io.readJSON(join(process.cwd(), 'packages/gauntlet-solana-contracts/artifacts/schemas', `${name}`))
 }
 
-const getProgramId = (name: CONTRACT_LIST, version: string) => {
-  const rawPK = io.readJSON(join(process.cwd(), 'packages/gauntlet-solana-contracts/artifacts/programId', `${name}`))
-  return Keypair.fromSecretKey(Uint8Array.from(rawPK))
+const getProgramKeypair = (name: CONTRACT_LIST, version: string): Keypair | undefined => {
+  try {
+    const rawPK = io.readJSON(join(process.cwd(), 'packages/gauntlet-solana-contracts/artifacts/programId', `${name}`))
+    return Keypair.fromSecretKey(Uint8Array.from(rawPK))
+  } catch (e) {
+    logger.warn(`No program id keypair set for program ${name}`)
+    return
+  }
+}
+
+const getProgramId = (name: CONTRACT_LIST): PublicKey | undefined => {
+  const envNames = {
+    [CONTRACT_LIST.ACCESS_CONTROLLER]: 'ACCESS_CONTROLLER',
+    [CONTRACT_LIST.OCR_2]: 'OCR2',
+    [CONTRACT_LIST.DEVIATION_FLAGGING_VALIDATOR]: 'DEVIATION_FLAGGING_VALIDATOR',
+  }
+
+  try {
+    return new PublicKey(process.env[envNames[name]]!)
+  } catch (e) {
+    logger.warn(`No program id set for program ${name}`)
+    return
```

Comment:
Why swallow the error here instead of letting it go to bubble up?

Reply:
You're right